packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -349,24 +330,115 @@ const createHistoryService = ({ strapi }: { strapi: Core.Strapi }) => {
         );
       };
 
+      /**
+       * Get an object with two keys:
+       * - results: an array with the current values of the relations
+       * - meta: an object with the count of missing relations
+       */
+      // TODO: Move outside this function to utils
+      const buildMediaResponse = async (
+        values: { id: Data.ID }[]
+      ): Promise<{ results: any[]; meta: { missingCount: number } }> => {
+        return (
+          values
+            // Until we implement proper pagination, limit relations to an arbitrary amount
+            .slice(0, 25)
+            .reduce(
+              async (currentRelationDataPromise, entry) => {
+                const currentRelationData = await currentRelationDataPromise;
+
+                // Entry can be null if it's a toOne relation
+                if (!entry) {
+                  return currentRelationData;
+                }
+
+                const relatedEntry = await strapi.db
+                  .query('plugin::upload.file')
+                  .findOne({ where: { id: entry.id } });
+
+                if (relatedEntry) {
+                  currentRelationData.results.push(relatedEntry);
+                } else {
+                  // The related content has been deleted
+                  currentRelationData.meta.missingCount += 1;
+                }
+
+                return currentRelationData;
+              },
+              Promise.resolve({
+                results: [] as any[],
+                meta: { missingCount: 0 },
```

Comment:
probably same question as the other attributes, are media sanitized if using this format?

Reply:
I'm not sure...I don't think so. But I was actually under the impression there was nothing to sanitize for media. Are there fields on a media asset that could contain sensitive data (createdBy is not populated in the response)?

Reply:
Can not think of a concrete attribute that needs to be sanitized, but I would try to follow the rule of sanitizing everything by default, just in case

Reply:
derrick might haunt us, hitting us with a stick, during our dreams if we dont do so